commit eb2a7e1eb920cd9e44223607978344cf4b95d990
Author: Jonas Karlman <jonas@kwiboo.se>
Date:   Tue Jun 16 02:47:28 2015 +0200

    Resolve .plex.direct domains

diff --git a/libavformat/tcp.c b/libavformat/tcp.c
index 4eaf0a0..bd7b17f 100644
--- a/libavformat/tcp.c
+++ b/libavformat/tcp.c
@@ -48,6 +48,8 @@ static int tcp_open(URLContext *h, const char *uri, int flags)
     int timeout = 50;
     char hostname[1024],proto[1024],path[1024];
     char portstr[10];
+    size_t len;
+    char *c;

     av_url_split(proto, sizeof(proto), NULL, 0, hostname, sizeof(hostname),
         &port, path, sizeof(path), uri);
@@ -62,6 +64,17 @@ static int tcp_open(URLContext *h, const char *uri, int flags)
             timeout = strtol(buf, NULL, 10);
         }
     }
+    len = strlen(hostname);
+    if (len > 12 && !strcmp(hostname + len - 12, ".plex.direct")) {
+        for (c = hostname; *c; c++) {
+            if (*c == '-') {
+                *c = '.';
+            } else if (*c == '.') {
+                *c = '\0';
+                break;
+            }
+        }
+    }
     memset(&hints, 0, sizeof(hints));
     hints.ai_family = AF_UNSPEC;
     hints.ai_socktype = SOCK_STREAM;

